runtime/bcverify/staticverify.c

```diff
@@ -857,9 +857,9 @@ checkBytecodeStructure (J9CfrClassFile * classfile, UDATA methodIndex, UDATA len
 			}
 			info = &(classfile->constantPool[index]);
 			if (info->tag != CFR_CONSTANT_Methodref) {
-				if (((flags & BCT_MajorClassFileVersionMask) >= BCT_Java8MajorVersionShifted)
-				&& (bc != CFR_BC_invokevirtual) && (info->tag == CFR_CONSTANT_InterfaceMethodref)
-				) {
+				BOOLEAN checkFlag = J9_ARE_ANY_BITS_SET(flags, BCT_Unsafe) || ((flags & BCT_MajorClassFileVersionMask) >= BCT_Java8MajorVersionShifted);
```

Comment:
A name like "isJava8orLater" would be clearer.  I'd also put the Unsafe check second.
```suggestion
				BOOLEAN isJava8OrLater = ((flags & BCT_MajorClassFileVersionMask) >= BCT_Java8MajorVersionShifted) || J9_ARE_ANY_BITS_SET(flags, BCT_Unsafe);
```

Reply:
Agreed and updated against the suggestion above.